tools/checked-c-convert/ProgramInfo.cpp

```diff
@@ -42,7 +42,18 @@ PointerVariableConstraint::PointerVariableConstraint(const QualType &QT, uint32_
 	while (Ty->isPointerType()) {
 		// Allocate a new constraint variable for this level of pointer.
 		vars.insert(K);
-		CS.getOrCreateVar(K);
+		VarAtom * V = CS.getOrCreateVar(K);
+   
+    if (Ty->isCheckedPointerType()) {
+      if (Ty->isCheckedPointerPtrType()) {
+        // Constrain V so that it can't be either wild or an array.
+        CS.addConstraint(CS.createNot(CS.createEq(V, CS.getArr()))); 
+        CS.addConstraint(CS.createNot(CS.createEq(V, CS.getWild())));
+        ConstrainedVars.insert(K);
+      } else if (Ty->isCheckedPointerArrayType()) {
+        llvm_unreachable("unsupported!");
+      }
```

Comment:
This is only handling bounds-safe interfaces of _Ptr type.  You can also have bounds-safe interfaces of _Array_ptr type.   It's fine to break this into two separate commits but the description of the commit should be modified.